### ISSUE TITLE
development mode issue

### ISSUE THREAD

**Issue by sau4321** [hampoelz/Capacitor-NodeJS#26]
Can it be used in development mode with vite? I am running into issues where I am not receiving response from my node.js file to my capacitor app while in development mode also I am using capacitor 6

**Comment by hampoelz** [comment]
Yes, everything should work. Does the problem only exist in developer mode or always? And have you already looked at the [examples](https://github.com/hampoelz/Capacitor-NodeJS?tab=readme-ov-file#examples)? _(The examples use slightly older packages, but it should be possible to simply update them without breaking anything)_

**Comment by sayaa10** [comment]
//CAPACITOR CONFIG:
{
  "appId": "com.beatsy.musicapp",
  "appName": "Beatsy",
  "bundledWebRuntime": false,
  "npmClient": "npm",
  "webDir": "www",
  "plugins": {
    "SplashScreen": {
      "launchShowDuration": 0
    },"CapacitorNodeJS": {
      "nodeDir": "/nodejs"
    }
  },
  "server": {
  "url": "http://192.168.18.10:5173/",
  "cleartext": true
},
  "cordova": {}
}

//VITE CONFIG

import path from 'path';
import react from '@vitejs/plugin-react';


const SRC_DIR = path.resolve(__dirname, './src');
const PUBLIC_DIR = path.resolve(__dirname, './static');
const BUILD_DIR = path.resolve(__dirname, './www',);
export default async () => {

  return  {
    plugins: [
      react(),

    ],
    root: SRC_DIR,
    base: '',
    publicDir: PUBLIC_DIR,
    build: {
      outDir: BUILD_DIR,
      assetsInlineLimit: 0,
      emptyOutDir: true,
      rollupOptions: {
        treeshake: false,
      },
    },
    resolve: {
      alias: {
        '@': SRC_DIR,
      },
    },
    server: {
      host: true,
    },

  };
}

// static/nodejs/main.js
const { channel } = require('bridge');

// Listens to "msg-from-capacitor" from the Capacitor layer.
channel.addListener('msg-from-capacitor', message => {
  console.log('[Node.js] Message from Capacitor: ' + message);

  // Sends a message back to the Capacitor layer.
  channel.send('msg-from-nodejs', `Replying to the message '${message}'.`, 'And optionally add more arguments.');
});

//CLIENT
import { Page } from "framework7-react"
import { NodeJS } from 'capacitor-nodejs';
import { useEffect, useState } from 'react';

 const Downloads = () => {
  const [message, setMessage] = useState('Waiting for message...');

  useEffect(() => {
    // Listen for messages from Node.js
    NodeJS.addListener('msg-from-nodejs', event => {
      setMessage(event.args[0]); // Store first argument in state
    });

    // Initialize communication
    NodeJS.whenReady().then(() => {
      NodeJS.send({
        eventName: 'msg-from-capacitor',
        args: ['Hello from Downloads component!'],
      });
    });

    return () => NodeJS.removeAllListeners();
  }, []);

  return (
    <Page className="pl-[50px]">
      <div>
        <h2>Message from Node.js:</h2>
        <p>{message}</p>
      </div>
    </Page>
  )
}

export default Downloads;


I just see waiting for message in my android studio app

**Comment by sayaa10** [comment]
I want to get live changes while running my app live using 'npm run dev'
can I do that?

**Comment by sayaa10** [comment]
In the node/main.js from where does the bridge module gets imported? im confused a bit,

here is a breakdown of what I did,

I installed the capacitor/nodejs library 
they edited my viteconfig file
followed the steps then 
I edited the capacitor config file,
then I did npx cap sync,
then npx cap copy,
then as you can see for live changes I have added the vite development server url inside the capacitor config
then i created a communication system between the server and client
then opened the android project on android studio
then tried to receive message from the nodejs server
BUT it didn't work
So my question is If have to build my react project everytime using npm run build then it would be really hard for development purpose 
I'm so confused

**Comment by sayaa10** [comment]
ohh okay now I get it, this is kinda hard for live development purpose. Is there a way to create a nodejs server directly from the /android directory on some localhost:PORT and access it from my capacitor based react app?
